altair/widget/__init__.py

```diff
@@ -0,0 +1,11 @@
+import anywidget  # type: ignore
+import traitlets
+import pathlib
+
+_here = pathlib.Path(__file__).parent
+
+
+class ChartWidget(anywidget.AnyWidget):
+    _esm = _here / "static" / "index.js"
+    spec = traitlets.Unicode().tag(sync=True)
```

Comment:
Why isn't spec also a dict?

Reply:
Yeah, I think it could be. This was just copied from the blog post. 

At this point I'm only really looking for feedback on the the project layout. The Python `ChartWidget` and JavaScript `render` functions are going to be rewritten.

Reply:
Makes sense. I still feel this could live in ipyvega since there is nothing that is strictly Altair specific in the renderer but I don’t feel strong either way.

Reply:
My hope is, but please correct me if I'm wrong here, that by incorporating `anywidget` it is possible to add a new template or upgrade the current existing html/js-templates in https://github.com/altair-viz/altair/blob/master/altair/utils/html.py.

Reply:
Would it make sense to deprecate ipyvega when this is ready then? I'd think so.

Reply:
@manzt, what does it take to write an anywidget to a self-contained html file? I may be mis-remembering, but I thought that was something that you said was possible.

Reply:
> Would it make sense to deprecate ipyvega when this is ready then? I'd think so.

Yeah, I think this could cover the same use-cases as the `VegaWidget` in ipyvega. I'm not sure if the mimerenderer portion of ipyvega is covered by Altair right now.

Reply:
You should be able to use use the [`ipywidgets.embed.embed_minimal_html`](https://ipywidgets.readthedocs.io/en/latest/embedding.html#embeddable-html-snippet):

```python
chart_widget = ChartWidget(...)

from ipywidgets.embed import embed_minimal_html

embed_minimal_html("chart.html", views=[chart_widget], title="my chart")
```

But this doesn't seem to be working for any third-party widgets i've tested at the moment... I'll try to have a look and figure out what's going on.